Hw8/Hw8/SkipList.cs

```diff
@@ -0,0 +1,357 @@
+
+namespace Hw8;
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Net.Http.Headers;
+using System.Security;
+
+/// <summary>
+/// Implementation of Skip List.
+/// </summary>
+/// <typeparam name="T">The type of elements stored in the skip list. Must implement IComparable&lt;T&gt; for ordering.</typeparam>
+public class SkipList<T> : IList<T>
+    where T : IComparable<T>
+{
+    private readonly int maxHeight = 16;
+    private Node head;
+    private Node tail;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SkipList&lt;T&gt"/> class.
+    /// </summary>
+    public SkipList()
+    {
+        this.head = new Node();
+        this.tail = new Node();
+        this.Count = 0;
+        for (int i = 0; i < this.maxHeight; i++)
+        {
+            this.head.Next.Add(this.tail);
+        }
+    }
+
+    /// <summary>
+    /// Gets a number of elements in skip list.
+    /// </summary>
+    public int Count { get; private set; }
+
+    /// <summary>
+    /// Gets a value indicating whether skip list is readonly.
+    /// </summary>
+    public bool IsReadOnly => false;
+
+    /// <summary>
+    /// Gets element on specifies index.
+    /// </summary>
+    /// <param name="index"> Index of element. </param>
+    /// <returns> Value of element with given index. </returns>
+    /// <exception cref="ArgumentOutOfRangeException"> index is not a valid index in the SkipList&lt;T&gt. </exception>
+    /// <exception cref="NotSupportedException"> SkipList&lt;T&gt does not support setting values at specified indexes. </exception>
+    public T this[int index]
+    {
+        get
+        {
+            if (index >= this.Count)
+            {
+                throw new ArgumentOutOfRangeException();
+            }
+
+            Node currentNode = this.head;
+            for (int i = 0; i <= index; i++)
+            {
+                currentNode = currentNode.Next[0];
+            }
+
+            return currentNode.Value;
+        }
+
+        set => throw new NotSupportedException();
+    }
+
+    /// <summary>
+    /// Checks Whether SkipList&lt;T&gt contains given value.
+    /// </summary>
+    /// <param name="value"> Given value. </param>
+    /// <returns> Whether SkipList&lt;T&gt contains given value. </returns>
+    public bool Contains(T value)
+    {
+        Node currentNode = this.Find(value).CurrentNode;
+        return currentNode.Value.Equals(value);
+    }
+
+    /// <summary>
+    /// Adds new element to SkipList&lt;T&gt.
+    /// </summary>
+    /// <param name="value"> Value of an element to be added. </param>
+    public void Add(T value)
+    {
+        var found = this.Find(value);
+        Node currentNode = found.CurrentNode;
+        Node[] remember = found.Remember;
+
+        if (!currentNode.Value.Equals(value))
+        {
+            this.Count++;
+            Node newNode = new Node();
```

Comment:
```suggestion
            Node newNode = new();
```
Не пишите в объявлении один и тот же тип дважды